v2/emailpassword/troubleshooting/cors-issues.mdx

```diff
@@ -0,0 +1,219 @@
+
+### `methods` [optional]
+
+This configures the Access-Control-Allow-Methods CORS header. It expects one of two input types.
+
+- Comma-delimited string - `'GET,PUT,POST'`
+- Array - `['GET', 'PUT', 'POST']`
+
+This is important because the header tells the browser what HTTP methods are allowed to be used when making a cross-origin request. Supertokens in particular uses `GET`, `POST` and `PUT` for functions such as creating new accounts, signing in/out etc.
+
+### `credentials` [must have]
+
+This configures the `Access-Control-Allow-Credentials` CORS header. It expects a boolean.
+
+If you set this header to true, then the browser will allow cookies to be sent in cross-origin requests. This is important because SuperTokens, by default uses cookies for sessions in the browser.
+
+## Common CORS Issues
+<br/>
+
+### `[Error] Credentials flag is true, but Access-Control-Allow-Credentials is not "true".`
+
+This happens when the CORS credentials field is not set to true.
+
+We can remedy this issue by setting credentials to `true` in our cors setting.
+
+```tsx
+import SuperTokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+let app = express();
+
+// ...other middlewares
+
+app.use(
+  cors({
+    origin: "http://127.0.0.1:3000",
+    allowedHeaders: ["content-type", ...SuperTokens.getAllCORSHeaders()],
+    methods: ["GET", "PUT", "POST", "DELETE"],
+    //highlight-start
+    credentials: true, // Check this line
+    //highlight-end
+
+  })  
+);
+```
+
+<br/>
+
+### `[Error] Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204`
+
+This happens when the CORS origin field is incorrect. Another varient of this error is when the OPTIONS API returns a 404 error. This can happen cause of misconfigured API gateway or reverse proxy. It can also be caused by a misconfiguration in the CORS origin field. 
+
+We can remedy this issue by ensuring the origin field lines up with the API domain located in other parts of the code. I’ve highlighted them in bold below -
+
+```tsx
+import supertokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+
+supertokens.init({
+  supertokens: {
+    connectionURI: "...",
+  },
+
+  appInfo: {
+    appName: "...",
+    apiDomain: "http://localhost:3001", 
+    websiteDomain: "http://localhost:3000",
+  },
+   recipeList: [/* ... */],
+});
+
+const app = express();
+app.use(
+   cors({
+      //highlight-start
+      origin: "http://localhost:3000", // Check this line
+      //highlight-end
+      allowedHeaders: ["content-type", ...supertokens.getAllCORSHeaders()],
+      methods: ["GET", "PUT", "POST", "DELETE"],
+      credentials: true,
+   })
+);
+```
+
+<br />
+
+### `[Error] Origin http://localhost:3000 is not allowed by Access-Control-Allow-Origin. Status code: 200`
```

Comment:
this is the same as the previous point? The status code deosn't really matter.. so combine this point with the previous one, and then say that there are three reasons:
- Misconfigured origin
- CORS middleware after supertokens middleware
- OPTIONS API returns a 404 due to misconfig of proxy

Reply:
Gotcha, ok. Let me make that change.

Reply:
What do you think about - 
`[Error] Cannot use wildcard in Access-Control-Allow-Origin when credentials flag is true.`

The solution for this error is the same - to ensure that origin lines up with website domain.

Reply:
I think what you have done now is fine.